pkg/autodiscovery/providers/telemetry.go

```diff
@@ -0,0 +1,49 @@
+package providers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/providers/names"
+	pkgconfig "github.com/DataDog/datadog-agent/pkg/config"
+)
+
+type TelemetryProvider struct{}
+
+func NewTelemetryProvider() *TelemetryProvider {
```

Comment:
since it is a new Provider component, could you use the Component framework to provide access to the `pkgconfig.Datadog` instance?